### PR TITLE
Terrain System bugfixes

### PR DESCRIPTION
**Pull request by mbalfour-amzn**
Another round of bugfixes and unit tests:
* Created shape mocks in LmbrCentral and exposed them for tests in other gems to use
* Exposed terrain mocks for other gems to use
* Added support for height/normal samplers across areas
* Added more unit tests

### CODE REVIEW

**Review by sphrose — APPROVED** [review]

File: Gems/LmbrCentral/Code/Mocks/LmbrCentral/Shape/MockShapes.h [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+#include <gmock/gmock.h>
+
+#include <AzCore/Component/ComponentApplication.h>
+#include <LmbrCentral/Shape/BoxShapeComponentBus.h>
+#include <LmbrCentral/Shape/ShapeComponentBus.h>
+
+namespace UnitTest
+{
+    class MockBoxShapeComponentRequests : public LmbrCentral::BoxShapeComponentRequestsBus::Handler
```
Comment (cgalvan) [inline Gems/LmbrCentral/Code/Mocks/LmbrCentral/Shape/MockShapes.h:18]: Nitpick: `public LmbrCentral::BoxShapeComponentRequestsBus::Handler` on newline

File: Gems/LmbrCentral/Code/Mocks/LmbrCentral/Shape/MockShapes.h [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        {
+            LmbrCentral::BoxShapeComponentRequestsBus::Handler::BusDisconnect();
+        }
+
+        MOCK_METHOD0(GetBoxConfiguration, LmbrCentral::BoxShapeConfig());
+        MOCK_METHOD0(GetBoxDimensions, AZ::Vector3());
+        MOCK_METHOD1(SetBoxDimensions, void(const AZ::Vector3& newDimensions));
+    };
+
+    class MockShapeComponentRequests : public LmbrCentral::ShapeComponentRequestsBus::Handler
```
Comment (cgalvan) [inline Gems/LmbrCentral/Code/Mocks/LmbrCentral/Shape/MockShapes.h:36]: Nitpick: `public LmbrCentral::BoxShapeComponentRequestsBus::Handler` on newline

**Review by cgalvan — APPROVED** [review]
Nice thanks for doing this!

**Review by invertednormal — APPROVED** [review]

File: Gems/Terrain/Code/Source/TerrainSystem/TerrainSystem.cpp [hunk 3]
```diff
@@ -152,27 +152,59 @@ AZ::Vector2 TerrainSystem::GetTerrainHeightQueryResolution() const
     return m_currentSettings.m_heightQueryResolution;
 }
 
+void TerrainSystem::ClampPosition(float x, float y, AZ::Vector2& outPosition, AZ::Vector2& delta) const
```
[the remark this answers is not in the document]
Comment (mbalfour-amzn) [reply Gems/Terrain/Code/Source/TerrainSystem/TerrainSystem.cpp]: Renamed, and added comments!

**Pull request merged**